entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt

```diff
@@ -718,23 +718,37 @@ class EntityService(
         neo4jRepository.deleteEntity(entityId)
 
     @Transactional
-    fun deleteEntityAttribute(entityId: String, attributeName: String, contextLink: String): Boolean {
+    fun deleteEntityAttributes(entityId: String, attributeName: String, contextLink: String): Boolean {
         val expandedAttributeName = expandJsonLdKey(attributeName, contextLink)!!
 
         if (neo4jRepository.hasPropertyOfName(EntitySubjectNode(entityId), expandedAttributeName))
-            return neo4jRepository.deleteEntityProperty(entityId, expandedAttributeName) >= 1
+            return neo4jRepository.deleteEntityProperty(
+                subjectNodeInfo = EntitySubjectNode(entityId), propertyName = expandedAttributeName, deleteAll = true) >= 1
         else if (neo4jRepository.hasRelationshipOfType(
-            EntitySubjectNode(entityId),
-            expandedAttributeName.toRelationshipTypeName())
-        )
-            return neo4jRepository.deleteEntityRelationship(
-                entityId,
-                expandedAttributeName.toRelationshipTypeName()
-            ) >= 1
+                EntitySubjectNode(entityId), expandedAttributeName.toRelationshipTypeName()))
+            return neo4jRepository.deleteEntityRelationship(subjectNodeInfo = EntitySubjectNode(entityId),
+                relationshipType = expandedAttributeName.toRelationshipTypeName(), deleteAll = true) >= 1
 
         throw ResourceNotFoundException("Attribute $attributeName not found in entity $entityId")
     }
 
+    @Transactional
+    fun deleteEntityAttribute(entityId: String, attributeName: String, datasetId: URI?, contextLink: String): Boolean {
+        val expandedAttributeName = expandJsonLdKey(attributeName, contextLink)!!
+
+        if (neo4jRepository.hasPropertyInstance(EntitySubjectNode(entityId), expandedAttributeName, datasetId))

```

Comment:
This makes me wonder about one thing, couldn't we do a neo4jRepository.deleteEntityAttribute which tries to delete a relation OR a property and you simply check the result to see if it was destroyed .. ? this removes the checks before trying to delete.

Reply:
In the deleteEntityAttribute you will try first to delete it as a property, check the result and if no node was deleted try as Relationship ?
I don't think we will gain a lot here (since in some cases will end up with the same number of queries).

Reply:
No I meant to do it in a single query. Try to delete the attribute as a Property AND a Relationships

Reply:
The problem here is that we cannot do smth like that: 
```
OPTIONAL MATCH (propEntity:${subjectNodeInfo.label} { id: ${'$'}entityId })-[:HAS_VALUE]->(attribute:Property { name: ${'$'}propertyName })
OPTIONAL MATCH (relEntity:${subjectNodeInfo.label} { id: ${'$'}entityId })-[:HAS_OBJECT]->(attribute:Relationship)-[:${attributeName.toRelationshipTypeName()}]->()
```
```
OPTIONAL MATCH (attribute)-[:HAS_VALUE]->(propOfAttribute:Property)
WITH attribute, propOfAttribute
OPTIONAL MATCH (attribute)-[:HAS_OBJECT]->(relOfAttribute:Relationship)
DETACH DELETE attribute, propOfAttribute, relOfAttribute
```
We need to declare two variables property in the first match and relationship in the second one, which results the following cypher query:
```
OPTIONAL MATCH (propEntity:${subjectNodeInfo.label} { id: ${'$'}entityId })-[:HAS_VALUE]->(property:Property { name: ${'$'}propertyName })
OPTIONAL MATCH (relEntity:${subjectNodeInfo.label} { id: ${'$'}entityId })-[:HAS_OBJECT]->(relationship:Relationship)-[:${attributeName.toRelationshipTypeName()}]->()
```
Then in the delete query
```
OPTIONAL MATCH (property)-[:HAS_VALUE]->(propOfProperty:Property)
WITH property, propOfProperty
OPTIONAL MATCH (property)-[:HAS_OBJECT]->(relOfProperty:Relationship)

OPTIONAL MATCH (relationship)-[:HAS_VALUE]->(propOfRelationship:Property)
WITH relationship, propOfRelationship
OPTIONAL MATCH (relationship)-[:HAS_OBJECT]->(relOfRelationship:Relationship)

DETACH DELETE property, propOfProperty, relOfProperty, relationship, propOfRelationship, relOfRelationship
```
It returns property variable not found if we want to delete a relationship.